source/opt/loop_descriptor.h

```diff
@@ -65,13 +66,27 @@ class Loop {
   inline BasicBlock* GetHeaderBlock() { return loop_header_; }
   inline const BasicBlock* GetHeaderBlock() const { return loop_header_; }
 
+  void SetHeaderBlock(ir::BasicBlock* new_header) { loop_header_ = new_header; }
+
   // Returns the latch basic block (basic block that holds the back-edge).
   inline BasicBlock* GetLatchBlock() { return loop_continue_; }
   inline const BasicBlock* GetLatchBlock() const { return loop_continue_; }
+  void SetLatchBlock(ir::BasicBlock* new_continue) {
+    loop_continue_ = new_continue;
+  }
 
   // Returns the BasicBlock which marks the end of the loop.
   inline BasicBlock* GetMergeBlock() { return loop_merge_; }
   inline const BasicBlock* GetMergeBlock() const { return loop_merge_; }
+  void SetMergeBlock(ir::BasicBlock* new_merge) { loop_merge_ = new_merge; }
+
+  inline BasicBlock* GetConditionBlock() { return loop_condition_block_; }
```

Comment:
That's very specific to the current view of the loop, keep it internal to your pass